test/Microsoft.ML.Core.Tests/UnitTests/TestCustomTypeRegister.cs

```diff
@@ -192,7 +196,7 @@ public void RegisterTypeWithAttribute()
             Assert.Equal(tribeEnumerable[0].Merged.HandCount, tribe[0].One.HandCount + tribe[0].Two.HandCount);
 
             // Build prediction engine from the trained pipeline.
-            var engine = ML.Model.CreatePredictionEngine<AlienHero, SuperAlienHero>(model);
+            var engine = ML.Model.CreatePredictionEngine<AlienHero, SuperAlienHero>(modelSaved);
```

Comment:
It might be good to have this test both for the model before saving it, and for the model that is loaded from disk. Ideally both models should behave the same, but if a bug is introduced that only appears in one of both cases, it might be good to have both of the tests.

The `InlineDataAttribute` could become handy for this, as used by the `PermutationFeatureImportanceTests`:
https://github.com/dotnet/machinelearning/blob/cacc72ff10daa1f257ef84b1bf0a17599d76dacc/test/Microsoft.ML.Tests/PermutationFeatureImportanceTests.cs#L30-L51

Reply:
Fixed.

---
In reply to: [401428872](https://github.com/dotnet/machinelearning/pull/4989#discussion_r401428872) [](ancestors = 401428872)